src/Select/index.js

```diff
@@ -58,9 +58,16 @@ export const getItemByValue = (value, items) => {
  * @param {array} items downshift index `items`
  * @returns {Number} index of item
  */
-export const getItemIndex = ({ props }, items) => {
-  const values = items.map(({ props }) => props.value);
-  return values.indexOf(props.value);
+export const getItemIndex = (item, items) => {
+  let result = 0;
+  if (isAction(item)) {
+    result = items
+      .map(({ props }) => props.onSelect)
+      .indexOf(item.props.onSelect);
+  } else {
+    result = items.map(({ props }) => props.value).indexOf(item.props.value);
+  }
+  return result;
```

Comment:
Do you think this would be a touch more legible if expressed this way?
```suggestion
  // find by matching onSelect callback
  if (item?.props?.onSelect) {
    return items
      .map(({ props }) => props.onSelect)
      .indexOf(item.props.onSelect);
  } 
  // find by matching value
  return items.map(({ props }) => props.value).indexOf(item.props.value);
```
return should be -1 if any indexOf doesn't find a match, not 0... but `let result = 0;` should always get overridden anyways, right?

Reply:
I do like your style better, but I initialize `result = 0` because downshift needs the index to be in range.

A result of `-1` should never happen, but if for some reason there's a bug and we can't find an item index, downshift state should just use the first item (index `0`)